cylc/flow/task_events_mgr.py

```diff
@@ -1000,27 +1005,70 @@ def _process_message_submitted(self, itask, event_time):
                 self.data_store_mgr.delta_task_state(itask)
             itask.state.outputs.set_completion(TASK_OUTPUT_STARTED, True)
             self.data_store_mgr.delta_task_output(itask, TASK_OUTPUT_STARTED)
-            return
 
-        itask.set_summary_time('submitted', event_time)
-        job_d = get_task_job_id(itask.point, itask.tdef.name, itask.submit_num)
-        self.data_store_mgr.delta_job_time(job_d, 'submitted', event_time)
-        self.data_store_mgr.delta_job_state(job_d, TASK_STATUS_SUBMITTED)
-        # Unset started and finished times in case of resubmission.
-        itask.set_summary_time('started')
-        itask.set_summary_time('finished')
+        else:
+            itask.set_summary_time('submitted', event_time)
+            job_d = get_task_job_id(
+                itask.point,
+                itask.tdef.name,
+                itask.submit_num,
+            )
+            self.data_store_mgr.delta_job_time(job_d, 'submitted', event_time)
+            self.data_store_mgr.delta_job_state(job_d, TASK_STATUS_SUBMITTED)
+            # Unset started and finished times in case of resubmission.
+            itask.set_summary_time('started')
+            itask.set_summary_time('finished')
 
-        self.reset_inactivity_timer_func()
-        if itask.state.status == TASK_STATUS_PREPARING:
-            # The job started message can (rarely) come in before the submit
-            # command returns - in which case do not go back to 'submitted'.
-            if itask.state_reset(TASK_STATUS_SUBMITTED):
-                itask.state_reset(is_queued=False)
-                self.setup_event_handlers(
-                    itask, self.EVENT_SUBMITTED, f'job {self.EVENT_SUBMITTED}')
-                self.data_store_mgr.delta_task_state(itask)
-                self.data_store_mgr.delta_task_queued(itask)
-            self._reset_job_timers(itask)
+            self.reset_inactivity_timer_func()
+            if itask.state.status == TASK_STATUS_PREPARING:
+                # The job started message can (rarely) come in before the
+                # submit command returns - in which case do not go back to
+                # 'submitted'.
```

Comment:
OK, if this happens, then this branch does the wrong thing. It doesn't create job proxies until the submit message is received, at which point the job stays in the submitted state instead of running. But, time being of the essence right now!, I will approve and merge this and post a follow-up fix.